### DISCUSSION TITLE
[macos]: I have installed STM32CubeIDE. Do I also need MX?

### DISCUSSION THREAD

**Discussion by ubaldot** [ussserrr/stm32pio#119] [Q&A] [answered]
I am asking because in my understanding MX is embedded in IDE - and I have installed only the latter, for some reason MX won't install on Sonoma. From the IDE I selected a board and generate the configuration code,  but then when I run `stm32pio` I get the following error:

`ERROR    FileNotFoundError: [Errno 2] No such file or directory: '/Applications/STMicroelectronics/STM32CubeMX.app/Contents/MacOs/STM32CubeMX' `

In-fact, I have only **/Applications/STMicroelectronics/STM32CubeMX.app**
Any workaround for this? 

If it may help, here is the structure of the blank test project that I made with STM32CubeIDE

<img width="311" alt="image" src="https://github.com/ussserrr/stm32pio/assets/10845685/87e6404d-8277-49af-949b-fe58e7a54dc3">

**Comment by ussserrr** [comment]
In macOS, *.app applications is really just a folder of a special kind that the OS is displaying in a different way. The actual executable that is needed by the stm32pio to run to generate a code is located inside this .app folder. I've never used this new IDE though to tell you whether the embedded inside it CubeMX is suitable for code-generation.

You can browse any .app folder via Finder by selecting it and clicking "Show package contents" in context menu. The default stm32pio parameter (that you see in the error message you mention) can give you a hint where to locate the executable binary. Then you can override the default parameter value in stm32pio.ini config file.

**Comment by ubaldot** [comment]
Yes, I am aware of it. :) This is the content of STM32CubeIDE.app: 

![image](https://github.com/ussserrr/stm32pio/assets/10845685/1c1adeba-838a-4cfc-aefa-77422c61a194)

The only executable I see is that STM32CubeIDE but there is no trace of MX :( 
In the Application folder there is only an executable called ProjectImport and the rest is pretty much everything related to Eclipse. :(

**Comment by ubaldot** [comment]
FYI: I managed to install MX thanks to the answer given [here](https://community.st.com/t5/stm32cubemx-mcus/cubemx-won-t-install-on-macos-sonoma-14-2-1/td-p/631536). 

EDIT: I managed to run stm32pio but I noticed that I have Core/Src and Core/Inc that have the same content as the generated Src and Inc (e.g. there is a double main.c file). 

EDIT2: I was following the example but the "Other Toolchains (GPDSC)"  is no [longer supported](https://community.st.com/t5/stm32cubemx-mcus/stm32cubemx-why-can-i-not-select-some-options-in-the-quot/td-p/129649).
What shall we choose?

**Comment by ussserrr** [comment, marked as answer]
I guess the default one is STM32CubeIDE now (see #113)
